src/libponyc/pass/docgen.c

```diff
@@ -796,6 +985,8 @@ static void doc_entity(docgen_t* docgen, docgen_opt_t* docgen_opt, ast_t* ast)
 
   // Now we can write the actual documentation for the entity
   fprintf(docgen->type_file, "# %s", name);
+  add_source_code_link(docgen, ast);
```

Comment:
this source code link should appear after adding the type params after line 991

Reply:
Done.